### ISSUE TITLE
npm link x in directory without node_modules or package.json file

### ISSUE THREAD

**Issue by ORESoftware** [npm/npm#14713]
#### I'm opening this issue because:

  - [x ] npm is doing something I don't understand.

#### What's going wrong?

I ran `npm link x `(in my case` npm link suman`) in a directory without a `package.json` file or `node_modules` folder and the command did not fail. It looks like it went down / fell back to the user home directory, it gave me this output:

`/Users/Olegzandr/node_modules/suman -> /Users/Olegzandr/.nvm/versions/node/v7.0.0/lib/node_modules/suman -> /Users/Olegzandr/WebstormProjects/oresoftware/suman`

I think there should be a warning if it uses the user home directory for the results of the `npm link x` command

Let me know if this makes sense

**Comment by legodude17** [comment]
I think that there should be docs in general about how `npm` will look down the file system for a `package.json` to install with.

**Comment by ORESoftware** [comment]
YES absolutely, however, if that directory that NPM walks down to = process.env.HOME, then a warning should be logged IMO

**Comment by legodude17** [comment]
So if there is no `package.json` in your cwd then walk down the file tree, but warn if it reaches `$HOME` and wants to install there. Should it stop the walk if it gets to $HOME?

**Comment by ORESoftware** [comment]
Yeah, I think NPM already stops walking when it gets to $HOME. But if it does arrive at $HOME there should be a warning saying, we just installed node modules in your $HOME directory, hope you like it :)
